mmcv/transforms/utils.py

```diff
@@ -0,0 +1,132 @@
+# Copyright (c) OpenMMLab. All rights reserved.
+
+import functools
+import inspect
+import weakref
+from collections import defaultdict
+from collections.abc import Iterable
+from contextlib import contextmanager
+from typing import Callable, Union
+
+from .base import BaseTransform
+
+
+class cacheable_method:
+
+    def __init__(self, func):
+
+        # Check `func` is to be bound as an instance method
+        func_args = inspect.getfullargspec(func).args
+        if len(func_args) == 0 or func_args[0] != 'self':
+            raise TypeError(
+                '@cacheable_method should only be used to decorate '
+                'instance methods (the first argument is `self`).')
```

Comment:
Can we directly use the `inspect.ismethod(func)`?

Reply:
I am afraid we can't. This check happens before the function is bound as a method.